DOCSP-49759: Vector search index (#33)
mongodb/docs-django: source/model-data/indexes.txt

   - `SearchIndex <{+api+}ref/models/indexes/#searchindex>`__ class in the
     {+django-odm+} API documentation.

.. _django-indexes-vector-search:

Atlas Vector Search Indexes
~~~~~~~~~~~~~~~~~~~~~~~~~~~

Atlas Vector Search indexes allow you to query data based on its semantic meaning
rather than by keyword matches. You can integrate vector search with full-text search
queries and AI frameworks to support a range of use cases.

To create a vector search index, assign the ``indexes`` option in your model's
``Meta`` class to a ``VectorSearchIndex`` object. Pass the following arguments to the
``VectorSearchIndex()`` constructor:

- ``name``: *(Optional)* The name of your Atlas Vector Search index. If you do not
  specify this argument, {+framework+} automatically generates an index name.
- ``fields``: The fields you want to index. At least one must be a vector

>   field, represented by an array of ``FloatField`` or ``IntegerField`` values and an 

Reviewer: "and an array size" -> "with a size"
(It would be really helpful to get intersphinx working to https://django-mongodb-backend.readthedocs.io/ so you can have size link to ArrayField.size) If you need help, let me know.

Author: Unfortunately additional changes to the intersphinx links are stalled, I can't configure it myself - the docs platform team does that - and they have some higher priority work at the moment. I added a link to the API docs using the source constant instead

   - `SearchIndex <{+api+}ref/models/indexes/#searchindex>`__ class in the
     {+django-odm+} API documentation.

.. _django-indexes-vector-search:

Vector Search Indexes
~~~~~~~~~~~~~~~~~~~~~

> Vector Search indexes specify the behavior of an Atlas Vector Search, which allows

Reviewer 2: ```suggestion
Atlas Vector Search indexes allow
```
S: full product name and edit bc it seems duplicative to say AVS twice

you to query data based on its semantic meaning rather than by keyword matches.
You can integrate vector search with full-text search queries and AI frameworks to
support a range of use cases.

> To create a Vector Search index, assign the ``indexes`` option in your model's

Reviewer 2: ```suggestion
To create an Atlas Vector Search index, assign the ``indexes`` option in your model's
```
S: I would lowercase "vector search" or include full product name

``Meta`` class to a ``VectorSearchIndex`` object. Pass the following arguments to the
``VectorSearchIndex()`` constructor:

> - ``name``: *(Optional)* The name of your Vector Search index. If you do not

Reviewer 2: ```suggestion
- ``name``: *(Optional)* The name of your Atlas Vector Search index. If you do not
```
S: lowercase or full product name

  specify this argument, {+framework+} automatically generates an index name.
- ``fields``: The fields you want to index. At least one must be a vector
  field, represented by an array of ``FloatField`` or ``IntegerField`` values and an 
  array size.
- ``similarities``: The :atlas:`similarity function </atlas-vector-search/vector-search-type/#std-label-avs-similarity-functions>`
  to use. You can use the ``"cosine"``, ``"dotProduct"``, or ``"euclidean"``
  similarity function. Specify the function as a single string value or a list
  of values that assigns functions to individual vector fields.

The following example updates the ``Recipe`` model's ``Meta`` class to create

> a Vector Search index named ``"vector_search_idx"`` on the ``ratings`` vector field

Reviewer 2: ```suggestion
an Atlas Vector Search index named ``"vector_search_idx"`` on the ``ratings`` vector field
```

and the ``cook_time`` numeric field:

.. literalinclude:: /includes/model-data/indexes.py
   :start-after: start-vector-search
   :end-before: end-vector-search
   :language: python
   :copyable:
   :emphasize-lines: 3-9

.. tip::

>    To learn more about Vector Search queries and indexes, see the following resources:

Reviewer 2: ```suggestion
   To learn more about Atlas Vector Search queries and indexes, see the following resources:
```
